[vijay-shankaranand] iP

vijay-shankaranand opened the pull request:
# JavAI
> You should name a variable using the same care with which you name a first-born child. — Robert C. Martin

JavAI frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
1. double-click it.
1. add your tasks.
1. let it manage your tasks for you 😉

And it is **FREE!**

Features:

- [x] Managing todos
- [x] Managing deadlines
- [x] Managing events
- [x] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the main method:
```
public static void main(String[] args) {
        new JavAI("./src/main/txtFolder/JavAI.txt").run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

vijay-shankaranand committed:
add Level-0 requirements

vijay-shankaranand committed:
add level-1 changes

vijay-shankaranand committed:
beautify Level-1

vijay-shankaranand committed:
add level-2 iteration

vijay-shankaranand committed:
add level-3 changes

vijay-shankaranand committed:
add Level-4 changes

vijay-shankaranand committed:
add A-TextUiTesting changes

vijay-shankaranand committed:
add level-5 changes

vijay-shankaranand committed:
add level-6 changes

vijay-shankaranand committed:
add level-7 changes

vijay-shankaranand committed:
add level-7 changes

vijay-shankaranand committed:
Merge branch 'branch-Level-7'

vijay-shankaranand committed:
Merge branch 'master' of https://github.com/vijay-shankaranand/ip

# Conflicts:
#	src/main/java/JavAI.java

vijay-shankaranand committed:
Merge branch 'branch-Level-7' of https://github.com/vijay-shankaranand/ip into branch-Level-7

# Conflicts:
#	src/main/java/JavAI.java

vijay-shankaranand committed:
remove head components in level-7 code

vijay-shankaranand committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/JavAI.java

vijay-shankaranand committed:
add level-8 changes

vijay-shankaranand committed:
Merge branch 'branch-Level-8'

vijay-shankaranand committed:
add A-MoreOOP changes

vijay-shankaranand committed:
add A-Packages changes

vijay-shankaranand committed:
Merge remote-tracking branch 'origin/add-gradle-support'

vijay-shankaranand committed:
add A-Gradle changes

vijay-shankaranand committed:
add A-JUnit changes

vijay-shankaranand committed:
add Level-9 changes

vijay-shankaranand committed:
add A-JavaDoc changes

vijay-shankaranand committed:
add A-CodingStandard changes

vijay-shankaranand committed:
Merge branch 'branch-Level-9'

vijay-shankaranand committed:
Merge branch 'branch-A-JavaDoc'

vijay-shankaranand committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/JavAI/Parser.java
#	src/main/java/JavAI/Ui.java

joel-foo commented on src/main/java/JavAI/Deadline.java:
```diff
@@ -0,0 +1,34 @@
+        super(description);
+        this.endDate = LocalDate.parse(endDate);
+        DateTimeFormatter inputFormatter = DateTimeFormatter.ofPattern("HHmm");
+       this.endTime = LocalTime.parse(endTime, inputFormatter);
```
perhaps can fix the indentation for this line?

joel-foo commented on src/main/java/JavAI/Event.java:
```diff
@@ -0,0 +1,31 @@
+
+    @Override
+    public String toString() {
+
```
perhaps can remove this empty line? there's many such empty lines in other files too.

joel-foo commented on src/main/java/JavAI/Storage.java:
```diff
@@ -0,0 +1,152 @@
+ * The Storage class handles loading tasks from and writing tasks to a file.
+ */
+public class Storage {
+    String filePath;
```
perhaps can make this `private`?

joel-foo commented on src/main/java/JavAI/JavAIException.java:
```diff
@@ -0,0 +1,8 @@
+
+public class JavAIException extends Exception {
+    public JavAIException(String message) {
+
```
perhaps can remove this empty line?

joel-foo commented on src/test/java/JavAI/StorageTest.java:
```diff
@@ -0,0 +1,55 @@
+    }
+
+    @Test
+    void testLoad() throws IOException {
```
you may want to name your test functions more explicitly - whether you expect a success / exception to be thrown.

joel-foo commented on src/main/java/JavAI/Deadline.java:
```diff
@@ -0,0 +1,34 @@
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.LocalTime;
```
perhaps can add an empty line below this import statement?

emzm2023 commented on src/main/java/JavAI/Deadline.java:
```diff
@@ -0,0 +1,34 @@
+     *
+     * @param description The description of the deadline task.
+     * @param endDate The due date of the deadline task.
+     * @param endDate The due time of the deadline task.
```
Should this be endTime instead?

emzm2023 commented on src/main/java/JavAI/JavAI.java:
```diff
@@ -0,0 +1,62 @@
+        while (sc.hasNext()) {
+            String input = sc.nextLine();
+            if (input.equals("bye")) {
+                storage.taskListWriter(tasks.getTasksAsArrayList());
```
I like that your method names are clear and detailed

emzm2023 commented on src/main/java/JavAI/TaskList.java:
```diff
@@ -0,0 +1,88 @@
+        return tasks.size();
+    }
+
+    /**
+     * Adds a task to the TaskList.
+     *
+     * @param task The task to be added.
+     */
```
I like how your Javadoc comments are consistent and in line with the standard

emzm2023 commented on src/main/java/JavAI/TaskList.java:
```diff
@@ -0,0 +1,88 @@
+     * @return An ArrayList containing all the tasks.
+     */
+    public ArrayList<Task> getTasksAsArrayList() {
+
```
Perhaps can remove such empty lines?

emzm2023 commented on src/test/java/JavAI/StorageTest.java:
```diff
@@ -0,0 +1,55 @@
+public class StorageTest {
+
+    private Storage storage;
+    private final String FILE_PATH = "./src/main/txtFolder/JavAI.txt";
```
I like the naming of this constant

emzm2023 reviewed (commented):
Looks good overall, comments are also easy to understand and consistent with standard!

xCOLOURx commented on src/main/java/JavAI/Deadline.java:
```diff
@@ -0,0 +1,34 @@
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " ( by: " +
+                this.endDate.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + " " + this.endTime + " )";
```
Perhaps you could make this shorter and less complicated by computing the date format in the line before?

xCOLOURx commented on src/main/java/JavAI/Parser.java:
```diff
@@ -0,0 +1,143 @@
+     * @param ui The Ui instance for displaying messages and user interface.
+     * @throws JavAIException If there's an error in parsing or executing the command.
+     */
+    public void parse(String input, TaskList tasks, Ui ui) throws JavAIException {
```
Perhaps you could shorten this method by making use of more methods?

xCOLOURx commented on src/main/java/JavAI/Storage.java:
```diff
@@ -0,0 +1,152 @@
+     *
+     * @return An ArrayList of Task objects loaded from the file.
+     */
+    public ArrayList<Task> load() {
+
```
Perhaps you can shorter this method and also avoid deep nesting? But I like the newlines you use to divide the code and make it more visible.

xCOLOURx commented on src/main/java/JavAI/TaskList.java:
```diff
@@ -0,0 +1,88 @@
+     * @param index The index of the task to retrieve.
+     * @return The task at the specified index.
+     * @throws JavAIException If the index is out of bounds.
+     */
```
I like the newline between the description and the param that makes it clearer to follow along.

xCOLOURx commented on src/test/java/JavAI/StorageTest.java:
```diff
@@ -0,0 +1,55 @@
+package javai;
+
+import static org.junit.jupiter.api.Assertions.*;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import java.util.ArrayList;
+import java.io.*;
+import java.util.Scanner;
```
Could perhaps group the import statements and separate them out with newlines? Like all the junit together, and the utils together.

xCOLOURx reviewed (commented):
Overall, I found your code easy to follow and well formatted. Just a few nits to fix but nothing major.

vijay-shankaranand committed:
add A-CheckStyle changes

vijay-shankaranand committed:
Merge branch 'branch-A-CheckStyle'

vijay-shankaranand committed:
add Level-10 changes

vijay-shankaranand committed:
Merge branch 'branch-Level-10'

vijay-shankaranand committed:
add A-Assertions changes

vijay-shankaranand committed:
Add A-CodeQuality changes

parser() and load() methods were too long hence hence code quality is affected

Shrunk these methods by adding multiple layers of abstraction to comply with SLAP

This improve code readability and code quality

vijay-shankaranand committed:
Merge pull request #2 from vijay-shankaranand/branch-A-CodeQuality

Add A-CodeQuality changes

vijay-shankaranand committed:
Merge branch 'master' into branch-A-Assertions

# Conflicts:
#	src/main/java/JavAI/Parser.java
#	src/main/java/JavAI/Storage.java
#	src/main/txtFolder/JavAI.txt

vijay-shankaranand committed:
Merge pull request #1 from vijay-shankaranand/branch-A-Assertions

add A-Assertions changes

vijay-shankaranand committed:
add B-Reminder changes

vijay-shankaranand committed:
Update JavAI.txt

proto-aiken-13 commented on src/main/java/JavAI/Ui.java:
```diff
@@ -1,4 +1,4 @@
+package javai;
 
 /**
  * The Ui class handles user interface interactions and message display.
```
I like how you describe it, but a nitpick: You should start by using a verb (e.g. you can say handles user interface interactions and message display)

proto-aiken-13 commented on src/main/java/JavAI/Parser.java:
```diff
@@ -1,4 +1,4 @@
+package javai;
 
 /**
  * The Parser class is responsible for parsing user input and executing corresponding commands
```
Nice description, but remember to start using verbs e.g. Parses user inputs and executes corresponding commands

proto-aiken-13 reviewed (commented):
This was VERY late but so far so good! Your code was generally okay to follow (even in this iteration of it).

vijay-shankaranand committed:
Add UI.png

vijay-shankaranand committed:
Update README.md

vijay-shankaranand committed:
Update README.md

vijay-shankaranand committed:
Update README.md

vijay-shankaranand committed:
Update README.md

vijay-shankaranand committed:
add final touches
